File: src/Telebox/ClientApp/src/models/Recording.ts
import { IChannel, IEpgListing } from "./XStream";

export enum RecordingStatus {
  Scheduled = 0,
  Recording = 1,
  Recorded = 2,
  Errored = 3,
}

export interface IRecording {
  id: number;
  connectionId: number;
  name: string;
  streamId: number;
  epgId: number;
  channelName: string;
  startTime: string;
  endTime: string;
  status: RecordingStatus;
  fileName: string;
  createdAt: string;
}

export interface IScheduleRecordingRequest {
  name: string;
  streamId: number;
  epgId: number;
  channelName: string;
  startTime: string;
  endTime: string;
}

const epgListingToScheduleRecordingRequest = (
  channel: IChannel,
  epg: IEpgListing
): IScheduleRecordingRequest => {
  return {
    name: epg.title,
    streamId: Number.parseInt(channel.streamId),
    epgId: Number.parseInt(epg.id),
    channelName: channel.name,
    startTime: epg.start,
    endTime: epg.end,
  };
};
export { epgListingToScheduleRecordingRequest };

File: src/Telebox/Data/Recording.cs
namespace Telebox.Data;

public enum RecordingStatus
{
    Scheduled = 0,
    Recording = 1,
    Recorded = 2,
    Errored = 3
}

public class Recording
{
    public int Id { get; set; }
    
    public int ConnectionId { get; set; }

    public string Name { get; set; } = default!;
    
    public int StreamId { get; set; }
    
    public int EpgId { get; set; }

    public string ChannelName { get; set; } = default!;
    
    public DateTime StartTime { get; set; }
    
    public DateTime EndTime { get; set; }
    
    public RecordingStatus Status { get; set; }

    public string FileName { get; set; } = default!;
    
    public DateTime CreatedAt { get; set; }
}